File: examples/python-repo/ecommerce/domain/customer/customer.py
class Customer:
    def __init__(self, customer_id, name, email):
        if customer_id <= 0:
            raise ValueError("customer id must be positive")
        if not name or not name.strip():
            raise ValueError("customer name must not be blank")
        if not email or "@" not in email:
            raise ValueError("invalid email")

        self._id = customer_id
        self._name = name
        self._email = email
        self._shipping_address = None

    @property
    def id(self):
        return self._id

    @property
    def name(self):
        return self._name

    @property
    def email(self):
        return self._email

    @property
    def shipping_address(self):
        return self._shipping_address

    def update_name(self, name):
        if not name or not name.strip():
            raise ValueError("name must not be blank")
        # DELIBERATE: unused variable (Ruff F841)
        old_name = self._name
        self._name = name

    def update_email(self, email):
        if not email or "@" not in email:
            raise ValueError("invalid email")
        self._email = email

    def set_shipping_address(self, address):
        self._shipping_address = address

File: examples/java-repo/src/main/java/com/example/ecommerce/domain/customer/Customer.java
package com.example.ecommerce.domain.customer;

// DELIBERATE: unused import (PMD UnusedImports)
import java.util.Collections;

public class Customer {

    private final long id;
    private String name;
    private String email;
    private Address shippingAddress;

    public Customer(long id, String name, String email) {
        if (id <= 0) {
            throw new IllegalArgumentException("customer id must be positive");
        }
        if (name == null || name.isBlank()) {
            throw new IllegalArgumentException("customer name must not be blank");
        }
        if (email == null || !email.contains("@")) {
            throw new IllegalArgumentException("invalid email");
        }
        this.id = id;
        this.name = name;
        this.email = email;
    }

    public long getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public String getEmail() {
        return email;
    }

    public Address getShippingAddress() {
        return shippingAddress;
    }

    public void updateName(String name) {
        if (name == null || name.isBlank()) {
            throw new IllegalArgumentException("name must not be blank");
        }
        this.name = name;
    }

    public void updateEmail(String email) {
        if (email == null || !email.contains("@")) {
            throw new IllegalArgumentException("invalid email");
        }
        this.email = email;
    }

    public void setShippingAddress(Address address) {
        this.shippingAddress = address;
    }
}
